File: content/problems/2788-split-strings-by-separator/solutions/manual-scan-nonempty-segments/Solution.kt
class Solution {
    fun splitWordsBySeparator(words:List<String>,separator:Char):List<String>{
        val answer=mutableListOf<String>();
        for(word in words){
            var start=0;
            for(end in 0..word.length)if(end==word.length||word[end]==separator){
                if(end>start)answer.add(word.substring(start,end));
                start=end+1
            }
        };
        return answer
    }
}

File: content/problems/2788-split-strings-by-separator/solutions/manual-scan-nonempty-segments/solution.cpp
class Solution {
    public:vector<string> splitWordsBySeparator(vector<string>& words,char separator){
        vector<string> answer;
        for(string& word:words){
            int start=0;
            for(int end=0;end<=(int)word.size();++end)if(end==(int)word.size()||word[end]==separator){
                if(end>start)answer.push_back(word.substr(start,end-start));
                start=end+1;
            }
        }
        return answer;
    }
};
